nx/source/services/capmtp.c

```diff
@@ -0,0 +1,129 @@
+#include "service_guard.h"
+#include "kernel/tmem.h"
+#include "runtime/hosversion.h"
+#include "services/capmtp.h"
+#include "services/sm.h"
+
+static Service g_capmtpRoot;
+static Service g_capmtp;
+static TransferMemory g_tmem;
+static Event g_connectEvent, g_scanErrorEvent;
+
+static Result _capmtpOpenSession(Service *srv);
+static Result _capmtpOpen(u32 max_folders, u32 max_img, u32 max_vid, const uint_least16_t *other_name);
+static Result _capmtpClose(void);
+static Result _capmtpNoInEventOut(u32 id, Event* event, bool autoclear);
+
+NX_GENERATE_SERVICE_GUARD_PARAMS(capmtp, (void* mem, size_t size, u32 max_folders, u32 max_img, u32 max_vid, const uint_least16_t *other_name), (mem, size, max_folders, max_img, max_vid, other_name));
+
+Result _capmtpInitialize(void* mem, size_t size, u32 max_folders, u32 max_img, u32 max_vid, const uint_least16_t *other_name) {
```

Comment:
String parameters should never be anything other than utf-8. Please change this to `const char*` and handle the conversion internally.